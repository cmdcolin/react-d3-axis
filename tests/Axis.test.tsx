import React from 'react';
import { render } from '@testing-library/react';
import Axis, {
  TOP, RIGHT, BOTTOM, LEFT,
} from '../src/Axis';

test('Renders bottom-oriented', () => {
  const { container } = render(
    <svg>
      <Axis
        style={{ orient: BOTTOM }}
        range={[0, 1000]}
        values={[0, 250, 500]}
        position={d => d * 2}
        format={d => `${d}`}
      />
    </svg>,
  );
  expect(container).toMatchSnapshot();
});

test('Renders top-oriented', () => {
  const { container } = render(
    <svg>
      <Axis
        style={{ orient: TOP }}
        range={[0, 1000]}
        values={[0, 250, 500]}
        position={d => d * 2}
        format={d => `${d}`}
      />
    </svg>,
  );
  expect(container).toMatchSnapshot();
});

test('Renders left-oriented', () => {
  const { container } = render(
    <svg>
      <Axis
        style={{ orient: LEFT }}
        range={[0, 1000]}
        values={[0, 250, 500]}
        position={d => d * 2}
        format={d => `${d}`}
      />
    </svg>,
  );
  expect(container).toMatchSnapshot();
});

test('Renders right-oriented', () => {
  const { container } = render(
    <svg>
      <Axis
        style={{ orient: RIGHT }}
        range={[0, 1000]}
        values={[0, 250, 500]}
        position={d => d * 2}
        format={d => `${d}`}
      />
    </svg>,
  );
  expect(container).toMatchSnapshot();
});
