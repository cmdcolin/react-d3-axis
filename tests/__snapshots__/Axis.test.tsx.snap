// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders bottom-oriented 1`] = `
<div>
  <svg>
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      stroke-width="1"
      text-anchor="middle"
    >
      <path
        d="M0.5,6V0.5H1000.5V6"
        stroke="black"
      />
      <g
        opacity="1"
        transform="translate(0,0)"
      >
        <line
          stroke="black"
          x1="0.5"
          x2="0.5"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="black"
          x="0.5"
          y="9"
        >
          0
        </text>
      </g>
      <g
        opacity="1"
        transform="translate(500,0)"
      >
        <line
          stroke="black"
          x1="0.5"
          x2="0.5"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="black"
          x="0.5"
          y="9"
        >
          250
        </text>
      </g>
      <g
        opacity="1"
        transform="translate(1000,0)"
      >
        <line
          stroke="black"
          x1="0.5"
          x2="0.5"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="black"
          x="0.5"
          y="9"
        >
          500
        </text>
      </g>
    </g>
  </svg>
</div>
`;

exports[`Renders left-oriented 1`] = `
<div>
  <svg>
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      stroke-width="1"
      text-anchor="end"
    >
      <path
        d="M-6,0.5H0.5V1000.5H-6"
        stroke="black"
      />
      <g
        opacity="1"
        transform="translate(0,0)"
      >
        <line
          stroke="black"
          x2="-6"
          y1="0.5"
          y2="0.5"
        />
        <text
          dy="0.32em"
          fill="black"
          x="-9"
          y="0.5"
        >
          0
        </text>
      </g>
      <g
        opacity="1"
        transform="translate(0,500)"
      >
        <line
          stroke="black"
          x2="-6"
          y1="0.5"
          y2="0.5"
        />
        <text
          dy="0.32em"
          fill="black"
          x="-9"
          y="0.5"
        >
          250
        </text>
      </g>
      <g
        opacity="1"
        transform="translate(0,1000)"
      >
        <line
          stroke="black"
          x2="-6"
          y1="0.5"
          y2="0.5"
        />
        <text
          dy="0.32em"
          fill="black"
          x="-9"
          y="0.5"
        >
          500
        </text>
      </g>
    </g>
  </svg>
</div>
`;

exports[`Renders right-oriented 1`] = `
<div>
  <svg>
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      stroke-width="1"
      text-anchor="start"
    >
      <path
        d="M6,0.5H0.5V1000.5H6"
        stroke="black"
      />
      <g
        opacity="1"
        transform="translate(0,0)"
      >
        <line
          stroke="black"
          x2="6"
          y1="0.5"
          y2="0.5"
        />
        <text
          dy="0.32em"
          fill="black"
          x="9"
          y="0.5"
        >
          0
        </text>
      </g>
      <g
        opacity="1"
        transform="translate(0,500)"
      >
        <line
          stroke="black"
          x2="6"
          y1="0.5"
          y2="0.5"
        />
        <text
          dy="0.32em"
          fill="black"
          x="9"
          y="0.5"
        >
          250
        </text>
      </g>
      <g
        opacity="1"
        transform="translate(0,1000)"
      >
        <line
          stroke="black"
          x2="6"
          y1="0.5"
          y2="0.5"
        />
        <text
          dy="0.32em"
          fill="black"
          x="9"
          y="0.5"
        >
          500
        </text>
      </g>
    </g>
  </svg>
</div>
`;

exports[`Renders top-oriented 1`] = `
<div>
  <svg>
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      stroke-width="1"
      text-anchor="middle"
    >
      <path
        d="M0.5,-6V0.5H1000.5V-6"
        stroke="black"
      />
      <g
        opacity="1"
        transform="translate(0,0)"
      >
        <line
          stroke="black"
          x1="0.5"
          x2="0.5"
          y2="-6"
        />
        <text
          dy="0em"
          fill="black"
          x="0.5"
          y="-9"
        >
          0
        </text>
      </g>
      <g
        opacity="1"
        transform="translate(500,0)"
      >
        <line
          stroke="black"
          x1="0.5"
          x2="0.5"
          y2="-6"
        />
        <text
          dy="0em"
          fill="black"
          x="0.5"
          y="-9"
        >
          250
        </text>
      </g>
      <g
        opacity="1"
        transform="translate(1000,0)"
      >
        <line
          stroke="black"
          x1="0.5"
          x2="0.5"
          y2="-6"
        />
        <text
          dy="0em"
          fill="black"
          x="0.5"
          y="-9"
        >
          500
        </text>
      </g>
    </g>
  </svg>
</div>
`;
